diff --git a/node_modules/@evidence-dev/mssql/index.cjs b/node_modules/@evidence-dev/mssql/index.cjs
index 7360078..fa3e55a 100644
--- a/node_modules/@evidence-dev/mssql/index.cjs
+++ b/node_modules/@evidence-dev/mssql/index.cjs
@@ -107,23 +107,41 @@ const buildConfig = function (database) {
 	if (database.authenticationType === 'default') {
 		return credentials;
 	} else if (database.authenticationType === 'azure-active-directory-default') {
+		delete credentials.user;
+		delete credentials.password;
+		credentials.authentication = { type: 'azure-active-directory-default', options: {} };
 		return credentials;
 	} else if (database.authenticationType === 'azure-active-directory-access-token') {
-		credentials.options.token = database.attoken;
+		delete credentials.user;
+		delete credentials.password;
+		credentials.authentication = {
+			type: 'azure-active-directory-access-token',
+			options: { token: database.attoken }
+		};
 		return credentials;
 	} else if (database.authenticationType === 'azure-active-directory-password') {
-		credentials.options = {
-			userName: database.pwuname,
-			password: database.pwpword,
-			clientId: database.pwclientid,
-			tenantId: database.pwtenantid
+		delete credentials.user;
+		delete credentials.password;
+		credentials.authentication = {
+			type: 'azure-active-directory-password',
+			options: {
+				userName: database.pwuname,
+				password: database.pwpword,
+				clientId: database.pwclientid,
+				tenantId: database.pwtenantid
+			}
 		};
 		return credentials;
 	} else if (database.authenticationType === 'azure-active-directory-service-principal-secret') {
-		credentials.options = {
-			clientId: database.spclientid,
-			clientSecret: database.spclientsecret,
-			tenantId: database.sptenantid
+		delete credentials.user;
+		delete credentials.password;
+		credentials.authentication = {
+			type: 'azure-active-directory-service-principal-secret',
+			options: {
+				clientId: database.spclientid,
+				clientSecret: database.spclientsecret,
+				tenantId: database.sptenantid
+			}
 		};
 		return credentials;
 	}
